### PR TITLE
Remove outdated jsUri

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

This [jsUri](https://github.com/derek-watson/jsUri) library we've got sitting around in the `/vendor` directory is 11 years old and is no longer needed. Browsers have long since implemented a standardised API for natively parsing URLs.

:fire::fire::fire:

There's quite a few other bits of code touched on here which are also crying out to be deleted, but that'll have to wait for another day...

Soundtrack: [High Voltage](https://www.youtube.com/watch?v=R-FxmoVM7X4) :musical_note: 

#### What should we test?

Check the Taxons/Taxonomies pages in superadmin settings where product categories are managed. 

#### Release notes

Removed jsUri in favour of web-native URL parsing

Changelog Category: Technical changes

### CODE REVIEW

File: vendor/assets/javascripts/jsuri.js [hunk 1]
```diff
@@ -1,2 +0,0 @@
...
...ion(a){n(a);return this},w=function(a){return m().getParamValue(a)},x=function(a){return m().getParamValues(a)},y=function(a,b){arguments.length===2?m().deleteParam(a,b):m().deleteParam(a);return this},z=function(a,b,c){arguments.length===3?m().addParam(a,b,c):m().addParam(a,b);return this},A=function(a,b,c){arguments.length===3?m().replaceParam(a,b,c):m().replaceParam(a,b);return this},B=function(){var a="",b=function(a){return a!==null&&a!==""};b(f())?(a+=f(),f().indexOf(":")!==f().length-1&&(a+=":"),a+="//"):h()&&b(j())&&(a+="//"),b(i())&&b(j())&&(a+=i(),i().indexOf("@")!==i().length-1&&(a+="@")),b(j())&&(a+=j(),b(k())&&(a+=":"+k())),b(l())?a+=l():b(j())&&(b(m().toString())||b(n()))&&(a+="/"),b(m().toString())&&(m().toString().indexOf("?")!==0&&(a+="?"),a+=m().toString()),b(n())&&(n().indexOf("#")!==0&&(a+="#"),a+=n());return a},C=function(){return new Uri(B())};return{protocol:f,hasAuthorityPrefix:h,userInfo:i,host:j,port:k,path:l,query:m,anchor:n,setProtocol:o,setHasAuthorityPrefix:p,setUserInfo:q,setHost:r,setPort:s,setPath:t,setQuery:u,setAnchor:v,getQueryParamValue:w,getQueryParamValues:x,deleteQueryParam:y,addQueryParam:z,replaceQueryParam:A,toString:B,clone:C}},jsUri=Uri;
```
Comment (Matt-Yorkley) [inline vendor/assets/javascripts/jsuri.js:2]: :tada:

File: app/assets/javascripts/admin/spree/spree.js.coffee [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 class window.Spree
   # Helper function to take a URL and add query parameters to it
-  @url: (uri, query) ->
-    if uri.path == undefined
-      uri = new Uri(uri)
-    if query
-      $.each query, (key, value) ->
-        uri.addQueryParam(key, value)
+  @url: (uri) ->
+    console.log(uri)
```
Comment (jibees) [inline app/assets/javascripts/admin/spree/spree.js.coffee:4]: Should we keep this `console.log` ?

**Review by jibees — APPROVED** [review]

File: app/assets/javascripts/admin/spree/spree.js.coffee (hunk 2 above)
Comment (Matt-Yorkley) [reply app/assets/javascripts/admin/spree/spree.js.coffee:4]: No :see_no_evil: 

Nice catch! :heart:

**Review by mkllnk — APPROVED** [review]
Shall we give the taxon screen a quick test?

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,

After this PR the Taxons/Taxonomies pages in superadmin settings work as before.
As superadmin:
- created a new taxonomy
- created child taxonomies
As hub:
- edited product and attributed newly created taxonomy
As customer:
- verified this appears in the product and shop description

All good!

**Pull request merged**